### DISCUSSION TITLE
could not call Module.put_attribute/3 because the module Plausible.Repo.Migrations.ChangeAllowedEventPropsType is already compiled

### DISCUSSION THREAD

**Discussion by drlogout** [plausible/analytics#4396] [Self-Hosted Support]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

When upgrading from 2.0.0 to 2.1.0 like described here https://github.com/plausible/analytics/discussions/4125 the migration fails with the error `could not call Module.put_attribute/3 because the module Plausible.Repo.Migrations.ChangeAllowedEventPropsType is already compiled`.

```bash
plausible-1            | Creation of Db successful!
plausible-1            | Loading plausible..
plausible-1            | Starting dependencies..
plausible-1            | Starting repos..
plausible-1            | Running migrations for Elixir.Plausible.Repo
plausible-1            | ** (ArgumentError) could not call Module.put_attribute/3 because the module Plausible.Repo.Migrations.ChangeAllowedEventPropsType is already compiled
plausible-1            |     (elixir 1.16.0) lib/module.ex:2360: Module.assert_not_readonly!/2
plausible-1            |     (elixir 1.16.0) lib/module.ex:2041: Module.__put_attribute__/5
plausible-1            |     lib/plausible-0.0.1/priv/repo/migrations/20230925072840_plugins_api_tokens.exs:2: (module)
plausible-1 exited with code 1
plausible-1            | Loading plausible..
plausible-1            | Starting dependencies..
plausible-1            | Starting repos..
plausible-1            | create Plausible.Repo database if it doesn't exist
plausible-1            | create Plausible.IngestRepo database if it doesn't exist
```

### Expected behavior

The migration should run without an error.

### Screenshots

![ScreenShot 2024-07-24 at 09 18 11@2x](https://github.com/user-attachments/assets/5a296371-4898-41d8-b20b-277f029ab7b8)

### Environment

```markdown
- OS: MacOS
```

**Comment by ruslandoga** [comment]
👋 @drlogout 

> OS: MacOS

Could you please share more information about your setup and the steps that lead to the error? If you are using Docker Desktop, can you also try reproducing it on a Linux host?
